src/PreallocatedLinearOperators.jl

```diff
@@ -0,0 +1,200 @@
+export PreallocatedLinearOperator
+
+abstract type PreallocatedAbstractLinearOperator{T,F1,F2,F3} <: AbstractLinearOperator{T,F1,F2,F3} end
+
+"""
+Type to represent a linear operator with preallocation. Implicit modifications may
+happen if used without care:
+```
+op = PreallocatedLinearOperator(rand(5, 5))
+v  = rand(5)
+x = op * v        # Uses internal storage and passes pointer to x
+y = op * ones(5)  # Stores on the same memory as x.
+y === x           # true. op * v is lost
+
+x = op * v        # Uses internal storage and passes pointer to x
+y = op * x        # Breaks! Equivalent to mul!(x, A, x)
+y == zeros(5)     # true. op * v and op * x are lost
```

Comment:
This should be clearer. If `y = op * x` breaks, how is the next instruction executed? What breaks exactly?